### DISCUSSION TITLE
Question UX Forms: * vs (Optional)

### DISCUSSION THREAD

**Discussion by alexookah** [scottyzen/woonuxt#204] [Q&A]
I was wondering: 
What do users prefer in forms to show a field is required vs optional? 

Should we label fields as 'required' (*) with  or just indicate 'optional' for the fields that are not required and assume the remaining fields are required by default?

@scottyzen

**Comment by alexookah** [comment]
ChatGPT:

> When designing forms or user interfaces, clarity is key. Most users prefer clear, explicit indications about which fields are required and which are optional. Here’s why:
> 
> Clear Instructions: Labeling required fields with an asterisk (*) provides immediate, clear guidance on what must be completed. This reduces the chance of users missing important fields and helps them complete the form correctly on the first try.
> 
> Reduced Assumptions: Assuming that all fields not marked as optional are required can lead to confusion or errors. Users might not realize which fields are actually mandatory if they’re not clearly marked, which can lead to frustration and incomplete submissions.
> 
> Consistency: Explicitly marking required fields and indicating optional ones helps maintain consistency and ensures users understand the form's requirements without having to guess.
> 
> Accessibility: Clear labels help users with disabilities, who may rely on screen readers or other assistive technologies. These tools often handle explicit labels better than implicit ones.
> 
> In summary, it’s generally better to label required fields with an asterisk (*) and mark optional fields explicitly. This approach minimizes ambiguity and makes the form-filling process smoother and more user-friendly.


Should we replace optional with empty and in required fields add * ?

**Comment by scottyzen** [comment]
I'd expect a checkout form to have mostly required fields so I think the (optional) label is a good fit here. For other larger forms that have a mix of both, the red asterisk is clearer. I'd like to hear what others think though.
